client/src/unifyfs-sysio.c

```diff
@@ -333,71 +335,83 @@ int UNIFYFS_WRAP(remove)(const char* path)
     }
 }
 
-int UNIFYFS_WRAP(stat)(const char* path, struct stat* buf)
+/* The main stat call for all the *stat() functions */
+static int __stat(const char* path, struct stat* buf)
 {
-    LOGDBG("stat was called for %s", path);
+    int gfid, fid;
+    unifyfs_file_attr_t fattr;
+    int ret;
 
-    if (unifyfs_intercept_path(path)) {
-        /* check that caller gave us a buffer to write to */
-        if (!buf) {
-            errno = EFAULT;
-            return -1;
-        }
+    gfid = unifyfs_generate_gfid(path);
```

Comment:
why not use `gfid = unifys_gfid_from_fid(fid)` after getting fid from path?

Reply:
IIRC, it was because the file may have a gfid but not a fid.  Like if you're stating '/unifycr'